stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/internal/MessageListItemViewTypeMapper.kt

```diff
@@ -52,7 +52,9 @@ internal object MessageListItemViewTypeMapper {
             message.deletedAt != null -> MESSAGE_DELETED
             message.isGiphyEphemeral() -> GIPHY
             containsGiphy -> GIPHY_ATTACHMENT
-            message.attachments.isNotEmpty() -> TEXT_AND_ATTACHMENTS
+            // TODO uncomment and sort order priority
+            // messageItem.message.attachments.isNotEmpty() -> TEXT_AND_ATTACHMENTS
+            messageItem.message.attachments.any { attachment -> attachment.hasLink() } -> LINK_ATTACHMENTS
```

Comment:
We're checking at the top `linksAndGiphy` and we could partition that into (giphy, links) add a check for links here instead of this.

This way we're just doing another check if there are any links, and even though the check will pass only if htere are links which are not giphy, we might be able to not have to go through all attachments and iterate but just filter out a smaller bit.

```
val (linksAndGiphy, _) = message.attachments.partition { attachment -> attachment.hasLink() }
val (links, giphy) = linksAndGiphy.partition { it.type == giphy } // whatever these checks have to be

val containsGiphy = giphy.isNotEmpty()
val containsLinsk = links.isNotEmpy()
```

Reply:
This could be a good alternative but I feel it depends on the real-world performance difference between `any()` and `partition()`.

To be clear I'm not leaning either way, it's just an interesting through.
I wonder if we could do a benchmark.

For now I will implementing per your suggestion.